src/app/beer-edit/beer-edit-form/beer-edit-form.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, EventEmitter, Input, Output, OnChanges } from '@angular/core';
+import { Location } from '@angular/common';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Beer } from '../../app.types';
+
+@Component({
+    selector: 'app-beer-edit-form',
+    templateUrl: './beer-edit-form.component.html',
+    styleUrls: ['./beer-edit-form.component.css']
+})
+export class BeerEditFormComponent implements OnChanges {
+    @Input() beer: Beer;
+    @Output() uploadRequest: EventEmitter<any> = new EventEmitter<any>();
+    @Output() submitRequest: EventEmitter<Beer> = new EventEmitter<Beer>();
+
+    public beerForm: FormGroup;
+
+    constructor (private location: Location, private fb: FormBuilder) {
+        this.beerForm = fb.group({
+            name: ['', Validators.required],
+            style: ['', Validators.required],
+            description: ['', Validators.required],
+            ingredients: '',
+            abv: ['', Validators.required],
+            epm: '',
+            brewery: ''
+        });
+    }
+    
+    ngOnChanges () {
+        this.beerForm.reset({
+            name: this.beer.name,
+            style: this.beer.style || '',
+            description: this.beer.description || '',
+            ingredients: this.beer.ingredients || '',
+            abv: this.beer.abv || '',
+            epm: this.beer.epm || '',
+            brewery: this.beer.brewery || ''
+        });
+    }
+
+    public submit ($event) {
```

Comment:
`$event: EVENT`?

Reply:
actually $event is unused, should be removed